velox/functions/prestosql/DataSizeFunctions.h

```diff
@@ -0,0 +1,446 @@
+  size_t valueLength = 0;
+  size_t decimalIndex = std::string::npos;
+
+  // Find the length of the numeric part and the index of the decimal point.
+  while (valueLength < decimalString.length() &&
+         (isdigit(decimalString[valueLength]) ||
+          decimalString[valueLength] == '.')) {
+    if (decimalString[valueLength] == '.') {
+      decimalIndex = valueLength;
+    }
+    valueLength++;
+  }
+  // Calculate precision and scale.
+  precision = valueLength - (decimalIndex != std::string::npos ? 1 : 0);
+  scale = (decimalIndex != std::string::npos) ? (valueLength - decimalIndex - 1)
+                                              : 0;
+}
+
+// TODO - how to create Long Decimal to store toDecimalValue result, this is a
+// private constructor
+template <typename P1, typename S1>
+inline LongDecimal<P1, S1> parseDecimal(
+    const std::string& dataSize,
+    const std::string& str,
+    const size_t& precision,
+    const size_t& scale) {
+  try {
+    LongDecimal<P1, S1> result;
+    // todo handle status error
+    Status status = toDecimalValue(StringView(str), precision, scale, result);
+    if (status.ok()) {
+      return result;
+    }
+    VELOX_USER_FAIL("Invalid data size: '{}'", dataSize);
+  } catch (const std::exception&) {
+    VELOX_USER_FAIL("Invalid data size: '{}'", dataSize);
+  }
+}
+
+template <typename P1, typename S1>
+inline LongDecimal<P1, S1> parseStringToDecimal(
+    const std::string& dataSize,
+    const std::string& str,
+    int precision,
+    int scale) {
+  try {
+    std::string value = dataSize.substr(0, precision + 1);
+    LongDecimal<P1, S1> result;
+    // todo handle status error
+    Status status = toDecimalValue(StringView(value), precision, scale, result);
+    if (status.ok()) {
+      return result;
+    }
+    VELOX_USER_FAIL("Invalid data size: '{}'", dataSize);
+  } catch (const std::exception&) {
+    VELOX_USER_FAIL("Invalid data size: '{}'", dataSize);
+  }
+}
+
+template <typename R, typename A, typename B>
+void decimalMultiplyFunction(R& out, const A& a, const B& b) {
+  out = checkedMultiply<R>(checkedMultiply<R>(R(a), R(b)), R(1));
+  DecimalUtil::valueInRange(out);
+}
+
+inline int128_t getDecimal(const std::string& decimalString) {
+  size_t precision = 0;
+  size_t scale = 0;
+  getPrecisionAndScale(decimalString, precision, scale);
+  VELOX_USER_CHECK_GT(precision, 0, "Invalid data size: '{}'", decimalString);
+  // TODO - Change parameters use calculated precision, scale. Using output
+  // parameters (38,0) here truncates information after decimal point.
+  std::string valueStr = decimalString.substr(0, precision + 1);
+  LongDecimal<P1, S1> value =
+      parseDecimal<P1, S1>(decimalString, valueStr, precision, scale);
+  // TODO - any better way to convert int128_t to LongDecimal<P1, S1> ?
+  Unit unit = parseUnit(decimalString, precision + 1);
+  int128_t unitValue = static_cast<int128_t>(unit);
+  std::string unitStr = boost::lexical_cast<std::string>(unitValue);
```

Comment:
@karteekmurthys I found DecimalUtil.toString() to convert `int128_t` to std::string in a Velox standard way. 

However, I did not find any function to extract precision and scale from a string.  There's [one](https://github.com/facebookincubator/velox/blob/main/velox/type/Type.cpp#L122) from short or long decimal